runtime/src/main/java/org/corfudb/runtime/clients/NettyClientRouter.java

```diff
@@ -191,6 +209,10 @@ public NettyClientRouter(@Nonnull NodeLocator node,
         outstandingRequests = new ConcurrentHashMap<>();
         shutdown = true;
 
+        sslAuthenticationFailure = new AtomicBoolean();
+        retryPolicy = new ExponentialBackoffRetry(null);
+        retryPolicy.setMaxRetryThreshold(maxRetryThreshold);
```

Comment:
Can you make the ExponentialBackoffRetry object immutable and pass the the threshold in the constructor ?